src/main/java/seedu/address/logic/commands/EditCommand.java

```diff
@@ -53,18 +53,17 @@ public class EditCommand extends Command {
     public static final String MESSAGE_NOT_EDITED = "At least one field to edit must be provided.";
     public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book.";
 
-    private final Index index;
+    private final Optional<Index> index;
     private final EditPersonDescriptor editPersonDescriptor;
 
     /**
      * @param index of the person in the filtered person list to edit
      * @param editPersonDescriptor details to edit the person with
      */
     public EditCommand(Index index, EditPersonDescriptor editPersonDescriptor) {
-        requireNonNull(index);
         requireNonNull(editPersonDescriptor);
 
-        this.index = index;
+        this.index = Optional.ofNullable(index);
         this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
     }
```

Comment:
maybe have another constructor so that we abstract out the null? that way we can still keep the `requireNonNull` check
```
public EditCommand(EditPersonDescriptor editPersonDescriptor) {
        requireNonNull(editPersonDescriptor);
        this.index = index;
        this.index = Optional.ofNullable(null);
        this.editPersonDescriptor = new EditPersonDescriptor(editPersonDescriptor);
}
```

Reply:
Agreed, thanks!